### PR TITLE
Move all remote dependencies to be local

### PR DESCRIPTION
**Pull request by gabek**
This change moves all of our remote dependencies to be local modules.  Hopefully there's no fallout from this.

@gingervitis @mattdsteele Let me know what you think!

## How to add a new Javascript/npm dependency with this setup

1. Go to `build/javascript`
1. Add your thing `npm install bloat.js`
1. Run `npm run build`

It'll then:

1. Download the javascript dependency
2. Use snowpack to do its modular magic
3. Copy the local files to `webroot/js/web_modules`

Note: These are not bundled, they're still standalone files.  If we want to bundle them for an additional performance boost we can address that with Snowpack as well.

### CODE REVIEW

**Comment by gabek** [comment]
Apparently this change to make things better makes things way worse.  So **shrug**.

Previously:

image

Now:

image

**Comment by mattdsteele** [comment]
Ah weird! I'll take a look at what might be going on.

**Comment by mattdsteele** [comment]
I think the big driver here is just that we aren't serving gzipped assets, and some of the files (especially tailwind.min.css) is pretty massive.

Here's what it looks like if I put Caddy in front and enable gzip:

image

By comparison, here's what's on `master`, which is pretty similar:

image

So #164 might be more important if we end up serving more files locally; or just expect folks to use a reverse proxy that does these optimizations.

**Comment by mattdsteele** [comment]
It's probably also worth looking at optimizing Tailwind specifically; even minified it's still hundreds of KB of render-blocking CSS.

Once this lands and we have a local JS toolchain in place, it should be possible to create a custom Tailwind build which only contains the styles we use. https://tailwindcss.com/docs/controlling-file-size

File: webroot/index-standalone-chat.html [hunk 1]
```diff
@@ -6,17 +6,16 @@
     <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet" />
```
Comment (gabek) [inline webroot/index-standalone-chat.html:6]: Missed changing to local copy of tailwind.

**Comment by gabek** [comment]
Thanks for looking at this Matt!  That all makes sense, and I feel much better now!  We can certainly look at fixing that bulky CSS.  I've wondered if we should be gzipping natively within our own web server, but ultimately decided against it as to not throw more CPU load at the whole system.  But maybe it's worth looking at.

**Pull request merged**